src/pkgcheck/checks/pkgdir.py

```diff
@@ -233,8 +272,23 @@ def feed(self, pkgset):
         if unknown:
             yield UnknownPkgDirEntry(sorted(unknown), pkg=pkg)
 
+        # we need to do Manifest separately, after we've got all the files
+        try:
+            manifest_stat = os.stat(pjoin(pkg_path, 'Manifest'))
+        except FileNotFoundError:
+            pass
+        else:
+            manifest_size_limit = (len(versions) *
+                ManifestSizeViolation.limit_per_version)
+            if manifest_stat.st_size > manifest_size_limit:
+                yield ManifestSizeViolation(manifest_stat.st_size,
+                                            manifest_size_limit,
+                                            len(versions),
+                                            pkg=pkg)
```

Comment:
Please move this check into separate function inside this `Check` class. This is easier to understand, and will make the short-circuit logic easier.
Then when you call the function, use something like `yield from self.<func-name>(...)`